### ISSUE TITLE
Can't read file from pandas library

### ISSUE THREAD

**Issue by chuongmep** [Apollo3zehn/PureHDF#53]
When I'm try read from pandas python, it return nothing. Whether it relate to schema version of HDFS ? 

Thank you 
```py
import numpy as np
import pandas as pd
#%pip install tables -U
import warnings
import os
import time
from tables import NaturalNameWarning
warnings.filterwarnings('ignore', category=NaturalNameWarning)
filePath =r"file.h5"
store = pd.HDFStore(filePath)
store.open()
group  = store.groups()
group
```
This is testing in cs: 
```cs
[Test]
    public void TestSaveHdf()
    {
        var file = new H5File()
        {
            ["my-group"] = new H5Group()
            {
                ["numerical-dataset"] = new double[] { 2.0, 3.1, 4.2 },
                ["string-dataset"] = new string[] { "One", "Two", "Three" },
                Attributes = new()
                {
                    ["numerical-attribute"] = new double[] { 2.0, 3.1, 4.2 },
                    ["string-attribute"] = new string[] { "One", "Two", "Three" }
                }
            }
        };
        file.Write("file.h5");
    }

    [Test]
    public void TestReadHdf()
    {
        // root group
        var file = H5File.OpenRead("file.h5");

// sub group
        var group = file.Group("my-group");


// dataset
        var dataset = group.Dataset("numerical-dataset");
        var datasetData = dataset.Read<double[]>();
        foreach (var item in datasetData)
        {
            Console.WriteLine(item);
        }
    }

```

**Comment by Apollo3zehn** [comment]
Thanks for your issue report. Maybe pandas does not support the newest hdf5 file layout. I will check today evening :-)

**Comment by Apollo3zehn** [comment]
Apart from that I think pandas uses a different HDF5 dataset layout (i.e. multidimensional dataset to represent a dataframe). Maybe you have more luck with H5py.

**Comment by chuongmep** [comment]
Thank you for your help, meaning the problem now is different version of hdf5 , I'm just confused because I'm extract data from C# and read data from python

**Comment by chuongmep** [comment]
Just provide more information H5py it work well, just small problem with encoder string . 
```py
# read hdf5 file
filePath =r"file3.h5"
import h5py
f = h5py.File(filePath, 'r')
list(f.keys())
# get dataset inside group
dataset = f['Category']
# get member inside dataset
list(dataset.keys())
member = dataset['table']
# show dataframe 
import pandas as pd
import numpy as np
arr = np.array(member)
print(arr.dtype)
# cast to string
df = pd.DataFrame(arr)
df
```
Cs Files : 
```cs
[Test]
    public void TestSaveDatatableToHdf()
    {
        string group = "Category";
        DataTable dataTable = new DataTable();
        dataTable.Columns.Add("Id", typeof(int));
        dataTable.Columns.Add("Name", typeof(string));
        dataTable.Columns.Add("Address", typeof(string));
        dataTable.Rows.Add(1, "Hoang", "Hanoi");
        dataTable.Rows.Add(2, "Chuong", "Hanoi");
        dataTable.Rows.Add(3, "Huy", "Hanoi");
        dataTable.Rows.Add(4, "Hieu", "Hanoi");


// Convert DataTable to array of TableRow
        TableRow[] array = dataTable.AsEnumerable()
            .Select(row => new TableRow
            {
                Id = row.Field<int>("Id"),
                Name = row.Field<string>("Name"),
                Address = row.Field<string>("Address")
            })
            .ToArray();

// Add to HDF using the compound data type
        var file = new H5File()
        {
            [group] = new H5Group()
            {
                ["table"] = array,
            }
        };

        file.Write("file3.h5");
    }
struct TableRow
{
    public int Id;
    public string Name;
    public string Address;
}
```

**Comment by Apollo3zehn** [comment]
This is the line where Pandas calls into pytables: 

https://github.com/pandas-dev/pandas/blob/84aca21d06574b72c5c1da976dd76f7024336e20/pandas/io/pytables.py#L1501

And here pytables filters out all groups that do not have the `pandas_type` attribute set:
https://github.com/pandas-dev/pandas/blob/84aca21d06574b72c5c1da976dd76f7024336e20/pandas/io/pytables.py#L1505

When you create a very simple pandas HDF5 file like this

```cs
import numpy as np
import pandas as pd
hdf = pd.HDFStore('hdf_file.h5')
df = pd.DataFrame(np.random.rand(5,3))
hdf.put('test', df)
```

and then open that file in e.g. HDFView, you`ll see how Pandas does store the data internally (and also the attribute `pandas_type` mentioned above):

![grafik](https://github.com/Apollo3zehn/PureHDF/assets/20972129/905b55ac-efb4-4eea-a2cd-b18d75da8bd9)

So I think you need to create a HDF5 file with a group/attribute/dataset structure that Pandas expects.

**Comment by chuongmep** [comment]
@Apollo3zehn , do you have any c# example can help with that ?

**Comment by Apollo3zehn** [comment]
@chuongmep I do not have any examples to create Pandas compatible HDF5 files via PureHDF. But as shown in my previous post you can easily create a test HDF5 file using Pandas and have a look into what Pandas expects to be present in that file. It does not look too difficult to mimic that format with PureHDF. 

Alternatively you could read the data produced by PureHDF into Python via `h5py` and then convert that dataset to a Pandas dataframe (I think this is what you did here: https://github.com/Apollo3zehn/PureHDF/issues/53#issuecomment-1893850100).

So to summarize: No, unfortunately, there is no "Pandas compatible" mode yet, but I will add it to my todo list.

**Comment by chuongmep** [comment]
Thank you for your help, that will useful for me !
